### PR TITLE
wrapping max tokens and temperature

### PR DESCRIPTION
**Pull request by EyalPaz-700**

### CODE REVIEW

File: llm_client/llm_api_client/aleph_alpha_client.py [hunk 1]
```diff
@@ -24,13 +24,14 @@ def __init__(self, config: LLMAPIClientConfig):
...
+    async def text_completion(self, prompt: str, model: str | None = None, max_tokens : int | None= None, temperature : float = 0 , **kwargs) ->\
             list[str]:
+        self._set_model_in_kwargs(kwargs, model)
         if max_tokens is None:
             raise ValueError("max_tokens must be specified")
-        self._set_model_in_kwargs(kwargs, model)
         kwargs[PROMPT_KEY] = prompt
-        kwargs[MAX_TOKENS_KEY] = max_tokens
+        kwargs["maximum_tokens"] = kwargs.pop("maximum_tokens", max_tokens)
+        kwargs["temperature"] = kwargs.pop("temperature", max_tokens)
```
Comment (uripeled2) [inline llm_client/llm_api_client/aleph_alpha_client.py:34]: this is a bug, should be kwargs["temperature"] = temperature

File: llm_client/llm_api_client/anthropic_client.py [hunk 2]
```diff
@@ -21,13 +20,15 @@ def __init__(self, config: LLMAPIClientConfig):
         self._headers[ACCEPT_HEADER] = ACCEPT_VALUE
         self._headers[AUTH_HEADER] = self._api_key
 
-    async def text_completion(self, prompt: str, model: str | None = None, max_tokens: int | None = None, **kwargs) ->\
+    async def text_completion(self, prompt: str, model: str | None = None, max_tokens: int | None = None, temperature: float = 0,
+                              **kwargs) -> \
             list[str]:
         if max_tokens is None:
             raise ValueError("max_tokens must be specified")
```
Comment (uripeled2) [inline llm_client/llm_api_client/anthropic_client.py:27]: add here if max_tokens is None or kwargs["max_tokens_to_sample"]: raise ValueError("max_tokens must be specified")

Comment (EyalPaz-700) [reply llm_client/llm_api_client/anthropic_client.py:27]: i think adding kwargs["max_tokens_to_sample"] is not the best , i would rather do : if max_tokens is None and kwargs["max_tokens_to_sample"] is None

Comment (uripeled2) [reply llm_client/llm_api_client/anthropic_client.py:27]: that is what I meant

**Pull request closed without merging**